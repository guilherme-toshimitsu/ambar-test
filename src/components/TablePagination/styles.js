const styles = theme => ({
  buttonsContainer: {
    display: 'flex',
    width: '100%',
  },
  previous: {
    textAlign: 'right',
    width: '46%',
  },
  current: {
    textAlign: 'center',
    width: '8%',
  },
  next: {
    textAlign: 'left',
    width: '46%',
  },
});

export default styles;
