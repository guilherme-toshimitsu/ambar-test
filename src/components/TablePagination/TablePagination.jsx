import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { TableCell, Button } from '@material-ui/core';

import styles from './styles';

const useStyles = makeStyles(styles, { name: 'TablePagination' });

const TablePagination = ({ count, page, rowsPerPage, onChangePage }) => {
  const classes = useStyles();

  return (
    <>
      {count > rowsPerPage && (
        <TableCell colSpan={7}>
          <div className={classes.buttonsContainer}>
            <div className={classes.previous}>
              {page > 0 && (
                <>
                  <Button data-testid={'table-pagination-first'} onClick={event => onChangePage(event, 0)}>
                    {'<<'}
                  </Button>
                  <Button data-testid={'table-pagination-prev-arrow'} onClick={event => onChangePage(event, page - 1)}>
                    {'<'}
                  </Button>
                  <Button data-testid={'table-pagination-prev'} onClick={event => onChangePage(event, page - 1)}>
                    {page}
                  </Button>
                </>
              )}
            </div>
            <div className={classes.current}>
              <Button data-testid={'table-pagination-current'} onClick={event => onChangePage(event, page)}>
                {page + 1}
              </Button>
            </div>
            <div className={classes.next}>
              {(page + 1) * rowsPerPage < count && (
                <>
                  <Button data-testid={'table-pagination-next'} onClick={event => onChangePage(event, page + 1)}>
                    {page + 2}
                  </Button>
                  <Button data-testid={'table-pagination-next-arrow'} onClick={event => onChangePage(event, page + 1)}>
                    {'>'}
                  </Button>
                  <Button
                    data-testid={'table-pagination-last'}
                    onClick={event => onChangePage(event, Math.floor(count / rowsPerPage))}
                  >
                    {'>>'}
                  </Button>
                </>
              )}
            </div>
          </div>
        </TableCell>
      )}
    </>
  );
};

TablePagination.defaultProps = {
  page: 0,
  count: 0,
  rowsPerPage: 10,
  onChangePage: () => {},
};

TablePagination.propTypes = {
  page: PropTypes.number,
  count: PropTypes.number,
  rowsPerPage: PropTypes.number,
  onChangePage: PropTypes.func,
};
export default TablePagination;
