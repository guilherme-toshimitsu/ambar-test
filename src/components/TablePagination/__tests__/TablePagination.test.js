import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TablePagination from '../';
import { Table, TableFooter, TableRow } from '@material-ui/core';

describe('<TablePagination> Route Component Loader', () => {
  it('renders', () => {
    const component = render(
      <Table>
        <TableFooter>
          <TableRow>
            <TablePagination count={27} page={0} rowsPerPage={10} onChangePage={() => {}} />
          </TableRow>
        </TableFooter>
      </Table>
    );
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders in a midle page', () => {
    const component = render(
      <Table>
        <TableFooter>
          <TableRow>
            <TablePagination count={27} page={1} rowsPerPage={10} onChangePage={() => {}} />
          </TableRow>
        </TableFooter>
      </Table>
    );
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders in a finalPage', () => {
    const component = render(
      <Table>
        <TableFooter>
          <TableRow>
            <TablePagination count={27} page={2} rowsPerPage={10} onChangePage={() => {}} />
          </TableRow>
        </TableFooter>
      </Table>
    );
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('simulate clicks', () => {
    const component = render(
      <Table>
        <TableFooter>
          <TableRow>
            <TablePagination count={27} page={1} rowsPerPage={10} onChangePage={() => {}} />
          </TableRow>
        </TableFooter>
      </Table>
    );

    const buttonFirst = component.getByTestId('table-pagination-first');
    const buttonBackArrow = component.getByTestId('table-pagination-prev-arrow');
    const buttonBack = component.getByTestId('table-pagination-prev');
    const buttonCurrent = component.getByTestId('table-pagination-current');
    const buttonNext = component.getByTestId('table-pagination-next');
    const buttonNextArrow = component.getByTestId('table-pagination-next-arrow');
    const buttonLast = component.getByTestId('table-pagination-last');

    fireEvent.click(buttonFirst);
    fireEvent.click(buttonBackArrow);
    fireEvent.click(buttonBack);
    fireEvent.click(buttonCurrent);
    fireEvent.click(buttonNext);
    fireEvent.click(buttonNextArrow);
    fireEvent.click(buttonLast);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('simulate clicks', () => {
    const component = render(
      <Table>
        <TableFooter>
          <TableRow>
            <TablePagination count={27} page={1} rowsPerPage={10} />
          </TableRow>
        </TableFooter>
      </Table>
    );

    const buttonFirst = component.getByTestId('table-pagination-first');
    fireEvent.click(buttonFirst);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
