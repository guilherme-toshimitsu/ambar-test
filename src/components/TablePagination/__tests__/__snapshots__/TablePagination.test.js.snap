// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination> Route Component Loader renders 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root"
  >
    <tfoot
      class="MuiTableFooter-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          class="MuiTableCell-root MuiTableCell-footer"
          colspan="7"
        >
          <div
            class="TablePagination-buttonsContainer-9"
          >
            <div
              class="TablePagination-previous-10"
            />
            <div
              class="TablePagination-current-11"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-current"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-next-12"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-last"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;&gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`<TablePagination> Route Component Loader renders in a finalPage 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root"
  >
    <tfoot
      class="MuiTableFooter-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          class="MuiTableCell-root MuiTableCell-footer"
          colspan="7"
        >
          <div
            class="TablePagination-buttonsContainer-139"
          >
            <div
              class="TablePagination-previous-140"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-first"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;&lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-current-141"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-current"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  3
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-next-142"
            />
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`<TablePagination> Route Component Loader renders in a midle page 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root"
  >
    <tfoot
      class="MuiTableFooter-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          class="MuiTableCell-root MuiTableCell-footer"
          colspan="7"
        >
          <div
            class="TablePagination-buttonsContainer-74"
          >
            <div
              class="TablePagination-previous-75"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-first"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;&lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-current-76"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-current"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-next-77"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  3
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-last"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;&gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`<TablePagination> Route Component Loader simulate clicks 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root"
  >
    <tfoot
      class="MuiTableFooter-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          class="MuiTableCell-root MuiTableCell-footer"
          colspan="7"
        >
          <div
            class="TablePagination-buttonsContainer-204"
          >
            <div
              class="TablePagination-previous-205"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-first"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;&lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-current-206"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-current"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-next-207"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  3
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-last"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;&gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`<TablePagination> Route Component Loader simulate clicks 2`] = `
<DocumentFragment>
  <table
    class="MuiTable-root"
  >
    <tfoot
      class="MuiTableFooter-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          class="MuiTableCell-root MuiTableCell-footer"
          colspan="7"
        >
          <div
            class="TablePagination-buttonsContainer-269"
          >
            <div
              class="TablePagination-previous-270"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-first"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;&lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &lt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-prev"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  1
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-current-271"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-current"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  2
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="TablePagination-next-272"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  3
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-next-arrow"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                data-testid="table-pagination-last"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  &gt;&gt;
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;
